tests/unit/test_derivative_utils.py

```diff
@@ -1,27 +1,110 @@
+import httpx
 import pandas as pd
 import pytest
 
 from bagel import mappings
 from bagel.utilities import derivative_utils
 
 
+def test_get_pipeline_from_backup_if_remote_fails(monkeypatch):
+    """
+    Test that the pipeline catalog is loaded from the local backup if the remote location is unreachable.
+    """
+    # TODO: Make a proper mock for the requests.get function
+    # or switch to httpx for better testing capabilities
+    nonsense_url = "https://does.not.exist.url"
+
+    def mock_httpx_get(*args, **kwargs):
+        response = httpx.Response(status_code=400, json={"key": "value"})
+        # This slightly odd construction is necessary to create a Response object
+        # that has the correct URL already baked in (I think), because otherwise we get the following
+        # RuntimeError: Cannot call `raise_for_status` as the request instance has not been set on this response.
+        # TODO: find a better solution or understand the problem better
+        response._request = httpx.Request("GET", nonsense_url)
+        return response
+
+    monkeypatch.setattr(httpx, "get", mock_httpx_get)
+
+    with pytest.warns(UserWarning) as w:
+        result = mappings.get_pipeline_catalog(
+            get_url=nonsense_url, get_path=mappings.PROCESSING_PIPELINE_PATH
```

Comment:
worth adding a note here that this test will fail if the submodule is not correctly initialized? since the test directly relies on the submodule